<a href="https://colab.research.google.com/github/srividya89/apps-project/blob/main/Srividya_S_(DL_project).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import tensorflow as tf 
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Input,Dense
import numpy as np


In [5]:
from sklearn.datasets import load_breast_cancer
Data=load_breast_cancer()


In [6]:
print(Data.DESCR)

.. _breast_cancer_dataset:

Breast cancer wisconsin (diagnostic) dataset
--------------------------------------------

**Data Set Characteristics:**

    :Number of Instances: 569

    :Number of Attributes: 30 numeric, predictive attributes and the class

    :Attribute Information:
        - radius (mean of distances from center to points on the perimeter)
        - texture (standard deviation of gray-scale values)
        - perimeter
        - area
        - smoothness (local variation in radius lengths)
        - compactness (perimeter^2 / area - 1.0)
        - concavity (severity of concave portions of the contour)
        - concave points (number of concave portions of the contour)
        - symmetry 
        - fractal dimension ("coastline approximation" - 1)

        The mean, standard error, and "worst" or largest (mean of the three
        largest values) of these features were computed for each image,
        resulting in 30 features.  For instance, field 3 is Mean Radius, f

In [7]:
features=Data.data 
target=Data.target
print(features.shape)
print(target.shape)


(569, 30)
(569,)


In [8]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test= train_test_split(features,target,test_size=0.2)
print(x_train.shape)
print(y_train.shape)
print(x_test.shape)
print(y_test.shape)


(455, 30)
(455,)
(114, 30)
(114,)


In [9]:
from keras.models import Sequential
from keras.layers import Dense


In [10]:
model=Sequential()
model.add(Dense(32,input_dim =30,activation ='relu'))
model.add(Dense(64,activation ='relu'))
model.add(Dense(1,activation='sigmoid'))


In [12]:
model.compile(loss='binary_crossentropy',optimizer='adam',metrics=['accuracy'])

In [13]:
model.summary()

Model: "sequential"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
dense (Dense)                (None, 32)                992       
_________________________________________________________________
dense_1 (Dense)              (None, 64)                2112      
_________________________________________________________________
dense_2 (Dense)              (None, 1)                 65        
Total params: 3,169
Trainable params: 3,169
Non-trainable params: 0
_________________________________________________________________


In [14]:
model.fit(x_train,y_train,epochs=10,batch_size=32)

Epoch 1/10
15/15 [==============================] - 12s 2ms/step - loss: 6.5228 - accuracy: 0.6178
Epoch 2/10
15/15 [==============================] - 0s 2ms/step - loss: 1.6095 - accuracy: 0.7960
Epoch 3/10
15/15 [==============================] - 0s 2ms/step - loss: 0.2869 - accuracy: 0.9244
Epoch 4/10
15/15 [==============================] - 0s 2ms/step - loss: 0.3939 - accuracy: 0.8668
Epoch 5/10
15/15 [==============================] - 0s 2ms/step - loss: 0.2410 - accuracy: 0.9193
Epoch 6/10
15/15 [==============================] - 0s 2ms/step - loss: 0.2369 - accuracy: 0.9224
Epoch 7/10
15/15 [==============================] - 0s 2ms/step - loss: 0.2697 - accuracy: 0.9172
Epoch 8/10
15/15 [==============================] - 0s 2ms/step - loss: 0.3267 - accuracy: 0.9020
Epoch 9/10
15/15 [==============================] - 0s 2ms/step - loss: 0.2299 - accuracy: 0.9259
Epoch 10/10
15/15 [==============================] - 0s 2ms/step - loss: 0.3479 - accuracy: 0.8790


In [15]:
model.fit(x_train,y_train,epochs=10)

Epoch 1/10
15/15 [==============================] - 0s 2ms/step - loss: 0.3679 - accuracy: 0.8813
Epoch 2/10
15/15 [==============================] - 0s 2ms/step - loss: 0.2656 - accuracy: 0.9055
Epoch 3/10
15/15 [==============================] - 0s 2ms/step - loss: 0.2267 - accuracy: 0.9209
Epoch 4/10
15/15 [==============================] - 0s 2ms/step - loss: 0.2918 - accuracy: 0.9055
Epoch 5/10
15/15 [==============================] - 0s 2ms/step - loss: 0.2372 - accuracy: 0.9187
Epoch 6/10
15/15 [==============================] - 0s 2ms/step - loss: 0.2268 - accuracy: 0.9231
Epoch 7/10
15/15 [==============================] - 0s 2ms/step - loss: 0.2258 - accuracy: 0.9099
Epoch 8/10
15/15 [==============================] - 0s 2ms/step - loss: 0.2596 - accuracy: 0.9033
Epoch 9/10
15/15 [==============================] - 0s 2ms/step - loss: 0.2881 - accuracy: 0.8967
Epoch 10/10
15/15 [==============================] - 0s 2ms/step - loss: 0.3445 - accuracy: 0.8989


In [16]:
scores=model.evaluate(x_test,y_test)

4/4 [==============================] - 0s 2ms/step - loss: 0.2884 - accuracy: 0.8772


In [17]:
print(scores)

[0.2883619964122772, 0.8771929740905762]


In [19]:
predictions=model.predict(x_test)
label=[]

In [20]:
for pred in predictions:
     if pred>=0.5:

       print("malignant")
     else:
       print("Benign")

Benign
Benign
malignant
Benign
Benign
malignant
Benign
Benign
malignant
malignant
malignant
malignant
Benign
Benign
malignant
Benign
Benign
Benign
malignant
malignant
Benign
malignant
malignant
malignant
malignant
malignant
Benign
Benign
Benign
malignant
Benign
Benign
Benign
malignant
malignant
malignant
malignant
Benign
malignant
Benign
malignant
malignant
malignant
Benign
Benign
malignant
Benign
Benign
malignant
Benign
malignant
Benign
Benign
malignant
malignant
malignant
malignant
malignant
Benign
Benign
Benign
malignant
Benign
malignant
Benign
malignant
malignant
malignant
malignant
Benign
Benign
malignant
Benign
malignant
malignant
malignant
malignant
malignant
Benign
malignant
Benign
Benign
Benign
malignant
malignant
malignant
malignant
Benign
Benign
malignant
malignant
malignant
malignant
malignant
Benign
malignant
malignant
malignant
malignant
Benign
Benign
malignant
malignant
malignant
Benign
Benign
malignant
malignant
malignant
Benign
Benign
Benign
Benign
malignant


model gives 87% accurate that the cancer is malignant